sources/logdog/src/error.rs

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+use std::io;
+use std::path::PathBuf;
+
+use snafu::{Backtrace, Snafu};
+
+/// Provides the list of errors for `logdog`.
+
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub(crate)")]
+pub(crate) enum Error {
+    #[snafu(display("Error creating the tarball file '{}': {}", path.display(), source))]
+    TarballFileCreate {
+        source: io::Error,
+        path: PathBuf,
+        backtrace: Backtrace,
+    },
+    #[snafu(display("Error writing to the tarball: {}", source))]
+    TarballWrite {
+        source: io::Error,
+        path: PathBuf,
```

Comment:
Some of these error variants aren't using the path in the message, would you please add it to improve the context?